### PR TITLE
Маршрут по тайным тропам

### PR DESCRIPTION
**Pull request by olegsorokin**
---
:mortar_board: [Маршрут по тайным тропам](https://up.htmlacademy.ru/react/9/user/30818/tasks/4)

### CODE REVIEW

File: project/src/components/private-route/private-route.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Navigate, RouteProps } from 'react-router-dom';
+import { AppRoute, AuthorizationStatus } from '../../const';
+
+type PrivateRouteProps = RouteProps & {
+  authorizationStatus: AuthorizationStatus;
+  children: JSX.Element;
+}
+
+function PrivateRoute(props: PrivateRouteProps): JSX.Element {
+  const { authorizationStatus, children } = props;
```
Comment (dianatsevan) [inline project/src/components/private-route/private-route.tsx:10]: `authorizationStatus` could be named in a more lightweight manner => `isAuth`, or `isAuthenticated`

File: project/src/components/private-route/private-route.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Navigate, RouteProps } from 'react-router-dom';
+import { AppRoute, AuthorizationStatus } from '../../const';
+
+type PrivateRouteProps = RouteProps & {
+  authorizationStatus: AuthorizationStatus;
```
Comment (dianatsevan) [inline project/src/components/private-route/private-route.tsx:5]: Why `AuthorizationStatus` type?

File: project/src/components/error/error.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Link } from 'react-router-dom';
+import { AppRoute } from '../../const';
+
+function Error() {
```
Comment (dianatsevan) [inline project/src/components/error/error.tsx:4]: Let's rename on `NotFoundPage`

File: project/src/const.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export enum AppRoute {
```
Comment (dianatsevan) [inline project/src/const.tsx:1]: Let's create folder `constants` where we will create specific constants' files, e.g. `src/constants/routs.ts`.

**Review by dianatsevan — CHANGES_REQUESTED** [review]

File: project/src/components/private-route/private-route.tsx (hunk 2 above)
Comment (olegsorokin) [reply project/src/components/private-route/private-route.tsx:5]: I found this realization example in a homework project:
https://github.com/htmlacademy-react/guess-melody-9/blob/6d3a170723b1e4028011899ab63c0f583b0d7473/project/src/components/private-route/private-route.tsx#L6

Comment (olegsorokin) [reply project/src/components/private-route/private-route.tsx:5]: > Why `AuthorizationStatus` type?

I found this realization example in a homework project:
https://github.com/htmlacademy-react/guess-melody-9/blob/6d3a170723b1e4028011899ab63c0f583b0d7473/project/src/components/private-route/private-route.tsx#L6

Comment (olegsorokin) [reply project/src/components/private-route/private-route.tsx:5]: I found this realization example in a homework project:
https://github.com/htmlacademy-react/guess-melody-9/blob/6d3a170723b1e4028011899ab63c0f583b0d7473/project/src/components/private-route/private-route.tsx#L6

**Review by dianatsevan — APPROVED** [review]

**Pull request merged**